src/graph.cc

```diff
@@ -554,24 +554,27 @@ bool ImplicitDepLoader::LoadDepFile(Edge* edge, const string& path,
     return false;
   }
 
-  // Preallocate space in edge->inputs_ to be filled in below.
-  vector<Node*>::iterator implicit_dep =
-      PreallocateSpace(edge, depfile.ins_.size());
-
-  // Add all its in-edges.
   for (vector<StringPiece>::iterator i = depfile.ins_.begin();
-       i != depfile.ins_.end(); ++i, ++implicit_dep) {
+       i != depfile.ins_.end(); ++i) {
     uint64_t slash_bits;
     if (!CanonicalizePath(const_cast<char*>(i->str_), &i->len_, &slash_bits,
                           err))
       return false;
 
     Node* node = state_->GetNode(*i, slash_bits);
-    *implicit_dep = node;
-    node->AddOutEdge(edge);
-    CreatePhonyInEdge(node);
+    depfile_deps->push_back(node);
   }
+  return true;
+}
 
+bool ImplicitDepLoader::LoadDepFile(Edge* edge, const string& path,
+                                    string* err) {
+  METRIC_RECORD("depfile load");
+  std::vector<Node*> depfile_deps;
+  if (!LoadDepfileDeps(edge, path, &depfile_deps, err))
+    return false;
+  if (!depfile_deps.empty())
+    ApplyLoadedDeps(edge, &depfile_deps[0], depfile_deps.size());
```

Comment:
std::vector<>::data()

But would you consider taking an iterator pair instead of a pointer-to-pointer, and size_t?

Reply:
The other callsite uses deps->nodes, deps->node_count; DepsLog::Deps has a good ol' plain array of pointers.

Reply:
Also, vector::data() is c++11 :(

Reply:
It is? Did not know that. Thanks for the insights. 

All the more reason that Ninja should switch to C++11 fully, but *shrug*.

Reply:
What's the advantage of std::vector<>::data()?

Reply:
Only that it's arguably nicer as &v[0] is unusual-looking.

Reply:
Actually I just found out that there's another important difference: It's always okay to call `data()` on an empty vector. See 95e0cb5156d295e0510b5296e441eecb1f8b1806